lib/facter/resolvers/virt_what.rb

```diff
@@ -13,7 +13,12 @@ def post_resolve(fact_name, _options)
         end
 
         def retrieve_from_virt_what(fact_name)
-          output = Facter::Core::Execution.execute('virt-what', logger: log)
+          command = if File.readable?('/opt/puppetlabs/puppet/bin/virt-what')
```

Comment:
If anything, you need to make sure it's executable
```suggestion
          command = if File.executable?('/opt/puppetlabs/puppet/bin/virt-what')
```